adaptive/runner.py

```diff
@@ -494,6 +496,19 @@ def __init__(
             def goal(_):
                 return False
 
+        if executor is None:
+            if isinstance(learner.function, types.LambdaType):
+                raise ValueError(
+                    "A lambda function cannot be pickled and "
+                    "therefore doesn't work with the default executor."
+                    "Either do not use a lamdba or use a framework that"
```

Comment:
```suggestion
                    "Either do not use a lamdba or use an executor that"
```

Reply:
Maybe we should give the specific example as a code snippet in the error message instead of just saying " `ipyparallel` with `dill`"